qm-plugins/qm-object-cache/html/class-qm-output-object-cache-ops.php

```diff
@@ -0,0 +1,158 @@
+	public function output() {
+		$data = $this->collector->get_data();
+		if ( empty( $data ) ) {
+			return;
+		}
+
+		$ops    = $data['operations'] ?? [];
+		$groups = $data['groups'] ?? [];
+
+		$this->before_tabular_output();
+
+		echo '<thead>';
+		echo '<tr>';
+		$this->output_filterable_table_col( __( 'Operation', 'qm-object-cache' ), array_keys( $ops ) );
+		$this->output_sortable_table_col( __( 'Key', 'qm-object-cache' ) );
+		$this->output_sortable_table_col( __( 'Size', 'qm-object-cache' ) );
+		$this->output_sortable_table_col( __( 'Time', 'qm-object-cache' ) );
+		$this->output_filterable_table_col( __( 'Group', 'qm-object-cache' ), array_keys( $groups ) );
+		$this->output_sortable_table_col( __( 'Result', 'qm-object-cache' ) );
+		echo '</tr>';
+		echo '</thead>';
+
+		echo '<tbody>';
+		foreach ( $ops as $op_name => $data ) {
+			foreach ( $data as $op ) {
+				echo '<tr data-qm-operation="' . esc_attr( $op_name ) . '" data-qm-group="' . esc_attr( $op['group'] ) . '">';
+				echo '<td class="qm-nowrap qm-ltr">' . esc_html( $op_name ) . '</td>';
+				echo '<td class="qm-nowrap qm-ltr">' . esc_html( $op['key'] ) . '</td>';
+				echo '<td class="qm-nowrap qm-ltr">' . esc_html( $this->process_size( $op['size'] ) ) . '</td>';
+				echo '<td class="qm-nowrap qm-ltr">' . esc_html( $this->process_time( $op['time'] ) ) . '</td>';
+				echo '<td class="qm-nowrap qm-ltr">' . esc_html( $op['group'] ) . '</td>';
+				echo '<td class="qm-nowrap qm-ltr">' . esc_html( $this->process_result( $op['result'] ) ) . '</td>';
+				echo '</tr>';
+			}
+		}
+		echo '</tbody>';
+
+		$this->after_tabular_output();
+	}
+
+	/**
+	 * @param array $class
+	 *
+	 * @return array
+	 */
+	public function admin_class( array $class ) {
+		$class[] = 'qm-object_cache_ops';
+		return $class;
+	}
+
+	/**
+	 * @param array<string, mixed[]> $menu
+	 * @return array<string, mixed[]>
+	 */
+	public function panel_menu( array $menu ) {
+		if ( isset( $menu['object_cache'] ) ) {
+			$menu['object_cache']['children'][] = $this->menu( array(
+				'id'    => 'qm-object_cache_ops',
+				'href'  => '#qm-object_cache_ops',
+				'title' => __( 'Operations', 'query-monitor' ),
+			));
+		}
+
+		return $menu;
+	}
+
+	/**
+	 * Returns in human readable size format
+	 *
+	 * @param $size int     Raw size
+	 * @return $size string Human readable size format
+	 */
+	public function process_size( int $size ) {
+		return size_format( $size, 2 );
+	}
+
+	/**
+	 * Returns in human readable time format.
+	 *
+	 * @param float $time   Raw time
+	 * @return string $time Human readable time format
+	 */
+	public function process_time( float $time ) {
+		return number_format_i18n( sprintf( '%0.1f', $time * 1000 ), 1 ) . 'ms';
+	}
+
+	/**
+	 * Returns in human readable result.
+	 *
+	 * @param $result  string Memcached operation result
+	 * @return $result string Human readable memcached operation result
+	 */
+	public function process_result( string $result ) {
+		switch ( trim( $result ) ) {
```

Comment:
I'm not sure how accurate this is going to be, cc @WPprodigy

Reply:
The description / rephrasing here is accurate. There's a couple types of operations where we don't leave these comments consistently, but that can be solved later on.